test/test_luaunit.lua

```diff
@@ -605,6 +605,27 @@ TestLuaUnitUtilities = { __class__ = 'TestLuaUnitUtilities' }
         lu.assertEquals( lu.private.patternFilter( {'!titi', '!toto'}, 'yyytotoxxx'), false )
         lu.assertEquals( lu.private.patternFilter( {'!titi', '!toto'}, 'tutu'), true )
         lu.assertEquals( lu.private.patternFilter( {'!titi', '!to..'}, 'yyytoxxx'), false )
+
+        -- combine patterns
+        lu.assertEquals( lu.private.patternFilter( { 'foo' }, 'foo'), true )
+        lu.assertEquals( lu.private.patternFilter( { 'foo', '!foo' }, 'foo'), false )
+        lu.assertEquals( lu.private.patternFilter( { 'foo', '!foo', 'foo' }, 'foo'), true )
+        lu.assertEquals( lu.private.patternFilter( { 'foo', '!foo', 'foo', '!foo' }, 'foo'), false )
+
+        lu.assertEquals( lu.private.patternFilter( { '!foo' }, 'foo'), false )
+        lu.assertEquals( lu.private.patternFilter( { '!foo', 'foo' }, 'foo'), true )
+        lu.assertEquals( lu.private.patternFilter( { '!foo', 'foo', '!foo' }, 'foo'), false )
+        lu.assertEquals( lu.private.patternFilter( { '!foo', 'foo', '!foo', 'foo' }, 'foo'), true )
+
+        lu.assertEquals( lu.private.patternFilter( { 'f..', '!foo', '__foo__' }, 'toto'), false )
+        lu.assertEquals( lu.private.patternFilter( { 'f..', '!foo', '__foo__' }, 'fii'), true )
+        lu.assertEquals( lu.private.patternFilter( { 'f..', '!foo', '__foo__' }, 'foo'), false )
+        lu.assertEquals( lu.private.patternFilter( { 'f..', '!foo', '__foo__' }, '__foo__'), true )
```

Comment:
This corresponds to a case where options were given that say "select `__foo__` and anything matching `f..`, but I don't want stuff that matches `foo`". I'd find it more natural to return `false` here.

It's debatable if a more specific `__foo__` should override the exclusion - but that would come with the big drawback of becoming dependent on the specific order of patterns...